prime-router/src/main/kotlin/tokens/Server2ServerAuthentication.kt

```diff
@@ -175,8 +175,8 @@ class Server2ServerAuthentication(val workflowEngine: WorkflowEngine) : Logging
         scope: String,
         jtiCache: JtiCache,
         actionHistory: ActionHistory? = null,
-    ): Boolean {
-        return try {
+    ) {
+        try {
```

Comment:
Does Kotlin have an either type?

Overall, there will likely only ever be the single caller outside of tests

Reply:
It doesn't in the standard library. You can bring in the [Arrow library](https://arrow-kt.io/) for a lot of functional goodies. I think the project could benefit from using some functional programming basics without going overboard but I understand it would be a group decision to bring in a new library.